src/halmos/__main__.py

```diff
@@ -101,6 +101,9 @@ def parse_args(args=None) -> argparse.Namespace:
     group_debug.add_argument(
         "--log", metavar="LOG_FILE_PATH", help="log every execution steps in JSON"
     )
+    group_debug.add_argument(
+        "--json", metavar="JSON_FILE_PATH", help="output test results in JSON"
```

Comment:
nit: the name of the flag isn't very descriptive, maybe `--json-output` or something like that?